tests/subsys/caf/sensor_manager/src/main.c

```diff
@@ -48,60 +42,75 @@ static void test_start(enum test_id test_id)
 	zassert_ok(err, "Test execution hanged");
 }
 
-static void test_event_manager(void)
+void *test_init(void)
+{
+	zassert_ok(app_event_manager_init(), "Error when initializing");
+	module_set_state(MODULE_STATE_READY);
+	test_start(TEST_SENSOR_INIT);
+	return NULL;
+}
+
+static void test_before(void *f)
 {
-	test_start(TEST_EVENT_MANAGER);
+	struct set_sensor_period_event *event_sensor1 = new_set_sensor_period_event();
+	struct set_sensor_period_event *event_sensor2 = new_set_sensor_period_event();
+	struct set_sensor_period_event *event_sensor3 = new_set_sensor_period_event();
+
+	event_sensor1->sampling_period = PRE_CHANGE_SAMPLING_PERIOD;
+	event_sensor1->descr = "Simulated sensor 1";
+	APP_EVENT_SUBMIT(event_sensor1);
+
+	event_sensor2->sampling_period = SAMPLING_PERIOD_LONG;
+	event_sensor2->descr = "Simulated sensor 2";
+	APP_EVENT_SUBMIT(event_sensor2);
+
+	event_sensor3->sampling_period = SAMPLING_PERIOD_LONG;
+	event_sensor3->descr = "Simulated sensor 3";
+	APP_EVENT_SUBMIT(event_sensor3);
```

Comment:
Make sure that test initialization is finished. For that:
- add event: test_initialization_done and send it here.
- In event processing - when test_initialization_done is received - set the semaphore.
- Here - wait for this semaphore (remember to check the waiting result).

This way we would be sure that no timing issue jumps in and when the test execution is started, all the configuration events are already processed.